File: py_solution/617. Merge Two Binary Trees.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def mergeTrees(self, t1, t2):
        """
        :type t1: TreeNode
        :type t2: TreeNode
        :rtype: TreeNode
        """
        
        ret= None
        
        if t1 and t2:
            ret = TreeNode(t1.val + t2.val)
            ret.left = self.mergeTrees(t1.left, t2.left)
            ret.right = self.mergeTrees(t1.right, t2.right)
        elif t1:
            ret = TreeNode(t1.val)
            ret.left = self.mergeTrees(t1.left, None)
            ret.right = self.mergeTrees(t1.right, None)
        elif t2:
            ret = TreeNode(t2.val)
            ret.left = self.mergeTrees(t2.left, None)
            ret.right = self.mergeTrees(t2.right, None)    
            
        return ret
            

File: c/617. Merge Two Binary Trees.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* mergeTrees(struct TreeNode* t1, struct TreeNode* t2) {
    if (!t1 && !t2){
        return NULL;
    }
    
    if (t1 && !t2) {
        return t1;
    }
    
    if (t2 && !t1) {
        return t2;
    }
    
    struct TreeNode* root = (struct TreeNode*)malloc(sizeof(struct TreeNode) * 1);
    root->val = t1->val + t2->val;
    root->left = mergeTrees(t1->left, t2->left);
    root->right = mergeTrees(t1->right, t2->right);
    
    return root;
}

